packages/nimble-components/src/table/types.ts

```diff
@@ -101,6 +101,26 @@ export interface TableRowSelectionEventDetail {
     selectedRecordIds: string[];
 }
 
+/**
+ * Event detail type for interactive column configuration changes.
+ */
+export interface TableColumnConfigurationChangeEventDetail {
```

Comment:
Can you capture some of the details (pun intended) of the event here as a comment?
In particular I'm thinking the when it's fired (only for user interactions that change column configuration) and also then array ordering (i.e. dom order)